File: AoC2016/JavaScript/day3.js
//puzzle setup
const fs = require('fs');
var input = fs.readFileSync('../day3in.txt','utf8');
input = input.split(/\r?\n/);

//do stuff
for (i in input) input[i] = input[i].trim().replace(/[ \t]{2,}/g,' ').split(' ').map(Number);
ct = 0;
for (i in input)
{
    let a = input[i][0];
    let b = input[i][1];
    let c = input[i][2];
    if (a + b > c && a + c > b && b + c > a) ct++;
}
console.log(ct);

File: AoC2020/Python3/day3.py
#puzzle setup
with open('./day3in.txt','r') as pInput:
  pInput = pInput.read().splitlines()

#do stuff
trees = x = y = 0
size = len(pInput[0])

while y < len(pInput):
  if pInput[y][x % size] == '#':
    trees += 1
  x += 3
  y += 1

print(trees)